org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/XMLLanguageServer.java

```diff
@@ -135,20 +141,23 @@ public void initialized(InitializedParams params) {
 	 * 
 	 * @param initializationOptionsSettings the XML settings
 	 */
-	public synchronized void updateSettings(Object initializationOptionsSettings) {
-		if (initializationOptionsSettings == null) {
+	public synchronized void updateSettings(Object initOptions) {
+		updateSettings(initOptions, true);
+	}
+
+	private synchronized void updateSettings(Object initOptions, boolean initLogs) {
+		if (initOptions == null) {
 			return;
 		}
 		// Update client settings
-		initializationOptionsSettings = AllXMLSettings.getAllXMLSettings(initializationOptionsSettings);
-		XMLGeneralClientSettings xmlClientSettings = XMLGeneralClientSettings
-				.getGeneralXMLSettings(initializationOptionsSettings);
+		Object initSettings = AllXMLSettings.getAllXMLSettings(initOptions);
+		XMLGeneralClientSettings xmlClientSettings = XMLGeneralClientSettings.getGeneralXMLSettings(initSettings);
 		if (xmlClientSettings != null) {
-			// Update logs settings
-			LogsSettings logsSettings = xmlClientSettings.getLogs();
-			if (logsSettings != null) {
-				LogHelper.initializeRootLogger(languageClient, logsSettings);
+			if (initLogs) {
+				// Update logs settings	
+				LogHelper.initializeRootLogger(languageClient, xmlClientSettings.getLogs());

```

Comment:
I think it worked like this before, but when user change `xml.logs.client` to false, it doesn't remove the LSP handler. In other words the user need to restart vscode.

Reply:
no https://github.com/eclipse/lemminx/pull/873/files#diff-8e86b1bafd539aecf46fde1e4f1cdcbaR43 unregisters all Log handlers, every time we change settings

Reply:
Oops sorry, I have not seen this piece of code. Thanks for the clarification.